### ISSUE TITLE
Find a solution for QStandardPaths

### ISSUE THREAD

**Issue by probonopd** [probonopd/linuxdeployqt#191]
__DO NOT USE__ [`QStringList QStandardPaths::standardLocations(QStandardPaths::AppDataLocation);`](http://doc.qt.io/qt-5/qstandardpaths.html)

According to the [Qt documentation](http://doc.qt.io/qt-5/qstandardpaths.html), this resolves to `"~/.local/share/<APPNAME>", "/usr/local/share/<APPNAME>", "/usr/share/<APPNAME>"` but clearly `/usr` is not where these things are located in an AppImage.

Right now, application authors need to change their code not to use `QStandardPaths::standardLocations` but use something like [ `QString QCoreApplication::applicationFilePath()`](http://doc.qt.io/qt-5/qcoreapplication.html#applicationDirPath) instead and construct a _relative_ path to `../share/kaidan/images/` from there.

As a result, it should work both in normal installations and in relocatable installations such as AppImages.

Either the behavior of __QStandardPaths__ should be changed in Qt to use paths relative to the Qt installation, or we need to find some other workaround.

**Comment by sgaist** [comment]
Hi,

Does applications/libraries have a way to know they are running from within an AppImage ?

**Comment by probonopd** [comment]
Yes, if the environment variables `$APPDIR` and `$APPIMAGE` are set, then it is safe to assume that the application is running within an AppImage.

This does not hold, however, when one uses the `--appimage-extract` command to extract the contents of the AppImage (e.g., useful for systems on which FUSE is not working). I would not recommend on relying on these variables for this reason, but to check for the existence of files relative to `bin/`. This way it should work inside the AppImage, also when the AppImage is extracted, and also when the application is installed traditionally.

Example:
https://github.com/Stellarium/stellarium/commit/43bc69948d91cd26785d4400f891dddf368a1465
